src/Models/UserModel.php

```diff
@@ -395,4 +395,9 @@ private function checkReturnType(): void
             throw new LogicException('Return type must be a subclass of ' . User::class);
         }
     }
+
+    public function getUserEntity(): User
```

Comment:
```suggestion
    /**
     * Returns User Entity Type
     *
     * @return class-string<User>
     */
    public function getReturnType(): string
```

Reply:
Is it better to add `createNewUser(array $data): User` or something instead of `getReturnType(): string`?

Reply:
That would omit the UserEntity entirely so we would still not be able to use UserEntity during the RegistrationAction?

Reply:
If we add the following method to `UserModel`:
```php
    /**
     * Returns a new User Entity.
     *
     * @param array<string, mixed> $data The data passed to the constructor
     */
    public function createNewUser(array $data = []): User
    {
        return new $this->returnType($data);
    }
```

`RegisterController` would be like this:
```php
    protected function getUserEntity(): User
    {
        $userProvider = $this->getUserProvider();

        return $userProvider->createNewUser();
    }
```

Reply:
@christianberkman 
I think adding a method to create a  new user is better than a method that just returns the user type string.
Would you like to make such an update?

Reply:
Yes, it would be better to open a fresh PR.

And I propose to add the first parameter:
```php
    /**
     * Returns a new User Entity.
     *
     * @param array<string, mixed> $data (Optional) user data
     */
    public function createNewUser(array $data = []): User
    {
        return new $this->returnType($data);
    }
```
Then, a dev can create a complete new User entity.
If a dev want to use *Complete Constructor* to the custom User entity, s/he needs complete data to construct a new entity.
Also, the CI4 Entity permits an empty instance creation, so we can just call it without arguments.

Reply:
I personally don't like using `mixed`, and I prefer to explicitly define the possible types as much as we can. For any other unexpected values, we can wait for bug reports from users.

Reply:
If we don't use `mixed`, something like this:
`array<string, array|bool|int|float|null|object|string>`

But it contains `array`, and it is not acceptable. Therefore,
`array<string, array<array-key, array|bool|int|float|null|object|string>|bool|int|float|null|object|string>`

But still contains `array`... Give up and use `mixed`.
`array<string, array<array-key, mixed>|bool|int|float|null|object|string>`

Which is better?
- `@param array<string, mixed> $data (Optional) user data`
- `@param array<string, array<array-key, mixed>|bool|int|float|null|object|string> $data (Optional) user data`

or something else?

Reply:
My choice is the second option because it strikes a balance between overly vague type definitions and overwhelming complexity. While it's not ideal, it offers a reasonable level of flexibility and clarity.
```php
@param array<string, array<array-key, mixed>|bool|int|float|null|object|string> $data (Optional) user data
```